libs/langchain-community/src/stores/message/dynamodb.ts

```diff
@@ -117,80 +158,129 @@ export class DynamoDBChatMessageHistory extends BaseListChatMessageHistory {
    * @returns Array of stored messages
    */
   async getMessages(): Promise<BaseMessage[]> {
-    const params: GetItemCommandInput = {
-      TableName: this.tableName,
-      Key: this.dynamoKey,
-    };
+    try {
+      const params: GetItemCommandInput = {
+        TableName: this.tableName,
+        Key: this.dynamoKey,
+      };
 
-    const response = await this.client.send(new GetItemCommand(params));
-    const items = response.Item
-      ? response.Item[this.messageAttributeName]?.L ?? []
-      : [];
-    const messages = items
-      .map((item) => ({
-        type: item.M?.type.S,
-        data: {
-          role: item.M?.role?.S,
-          content: item.M?.text.S,
-        },
-      }))
-      .filter(
-        (x): x is StoredMessage =>
-          x.type !== undefined && x.data.content !== undefined
-      );
-    return mapStoredMessagesToChatMessages(messages);
-  }
+      const response = await this.client.send(new GetItemCommand(params));
+      const items = response.Item
+        ? response.Item[this.messageAttributeName]?.L ?? []
+        : [];
+      const messages = items
+        .filter(
+          (
+            item
+          ): item is AttributeValue & { M: DynamoDBSerializedChatMessage } =>
+            item.M !== undefined
+        )
+        .map((item) => {
+          let data: {
+            role?: string;
+            content: string | undefined;
+            additional_kwargs?: Record<string, unknown>;
+          } = {
+            role: item.M?.role?.S,
+            content: item.M?.text.S,
+            additional_kwargs: item.M?.additional_kwargs?.S
+              ? JSON.parse(item.M?.additional_kwargs.S)
+              : undefined,
+          };
 
-  /**
-   * Deletes all messages from the DynamoDB table.
-   */
-  async clear(): Promise<void> {
-    const params: DeleteItemCommandInput = {
-      TableName: this.tableName,
-      Key: this.dynamoKey,
-    };
-    await this.client.send(new DeleteItemCommand(params));
+          return {
+            type: item.M?.type.S,
+            data: data,
+          };
+        })
+        .filter(
+          (x): x is StoredMessage =>
+            x.type !== undefined && x.data.content !== undefined
+        );
+      return mapStoredMessagesToChatMessages(messages);
+    } catch (error) {
+      console.error("Error getting messages:", error);
```

Comment:
Should we throw here?

Reply:
Thanks for the code review!
You're right, I fixed it.